extension/icu/icu-strptime.cpp

```diff
@@ -157,6 +159,77 @@ struct ICUStrptime : public ICUDateFunc {
 		bind = bound_function.bind;
 		bound_function.bind = StrpTimeBindFunction;
 	}
+
+	static bool CastFromVarchar(Vector &source, Vector &result, idx_t count, CastParameters &parameters) {
+		auto &cast_data = (CastData &)*parameters.cast_data;
+		auto info = (BindData *)cast_data.info.get();
+		CalendarPtr cal(info->calendar->clone());
+		const auto tz_name = info->tz_setting.c_str();
+
+		UnaryExecutor::ExecuteWithNulls<string_t, timestamp_t>(
+		    source, result, count, [&](string_t input, ValidityMask &mask, idx_t idx) {
+			    timestamp_t result;
+			    const auto str = input.GetDataUnsafe();
+			    const auto len = input.GetSize();
+			    string_t tz(nullptr, 0);
+			    if (!Timestamp::TryConvertTimestampTZ(str, len, result, tz)) {
+				    auto msg = Timestamp::ConversionError(string(str, len));
```

Comment:
Perhaps use `HandleCastError::AssignError` here instead of reimplementing the branching/throwing?